etl/fileNameConsumer.go

```diff
@@ -0,0 +1,43 @@
+package etl
+
+import (
+	"fmt"
+	"sync"
+)
+
+// FileNameConsumer consumes a channel of file names and produces an entry per line to the outgoing channel.
+type FileNameConsumer struct {
+	wg       sync.WaitGroup
+	incoming chan string
+	outgoing chan string
+}
+
+// Start causes the FileNameConsumer to begin consuming and processing items from the incoming channel,
+// asynchronously.
+func (fnc *FileNameConsumer) Start() {
+	go func() {
+		fnc.wg.Add(1)
+		defer fnc.wg.Done()
+
+		for filename := range fnc.incoming {
+			fnc.wg.Add(1)
+			go fnc.consume(filename)
+		}
+	}()
+}
+
+func (fnc *FileNameConsumer) consume(filename string) {
+	defer fnc.wg.Done()
+
+	fmt.Printf("FileNameConsumer: consumed file name %s\n", filename)
+
+	for i := range [32]int{} {
+		fnc.outgoing <- fmt.Sprintf("%s:%d", filename, i+1)
+	}
+}
+
+// Finish waits for all items to be consumed from the incoming channel.
+func (fnc *FileNameConsumer) Finish() {
+	fnc.wg.Wait()
+	close(fnc.outgoing)
```

Comment:
I would lift the closing of channels up into main, because that's where you're constructing them. Otherwise, what you see at your entrypoint is channels being created and then you gotta go hunt for where they're being closed.

Other thought is by lifting it out of Finish(), it's much easier to test because you've pulled the channel's state out of this struct.